src/services/workspace-client/devWorkspaceClient.ts

```diff
@@ -127,10 +133,54 @@ export class DevWorkspaceClient extends WorkspaceClient {
           ]
         },
         spec: pluginDevfile
-      });
+      };
+      devWorkspaceTemplates.push(theiaDWT);
+    }
 
-      this.addPlugin(createdWorkspace, pluginName, theiaDWT.metadata.namespace);
+    const devWorkspace: V1alpha2DevWorkspace = createdWorkspace;
+    // call theia library to insert all the logic
+    const inversifyBindings = new InversifyBinding();
+    const container = await inversifyBindings.initBindings({
+      pluginRegistryUrl: pluginRegistryUrl || '',
+      axiosInstance: this.axios,
+      insertTemplates: false,
+    });
+    const cheTheiaPluginsContent = optionalFilesContent['.che/che-theia-plugins.yaml'];
+    const vscodeExtensionsJsonContent = optionalFilesContent['.vscode/extensions.json'];
+    const cheTheiaPluginsDevfileResolver = container.get(CheTheiaPluginsDevfileResolver);
+
+    let sidecarPolicy: SidecarPolicy;
+    const devfileCheTheiaSidecarPolicy = (devfile as V1alpha2DevWorkspaceSpecTemplate).attributes?.['che-theia.eclipse.org/sidecar-policy'];
+    if (devfileCheTheiaSidecarPolicy === 'USE_DEV_CONTAINER') {
+      sidecarPolicy = SidecarPolicy.USE_DEV_CONTAINER;
+    } else {
+      sidecarPolicy = SidecarPolicy.MERGE_IMAGE;
     }
+    console.debug('Loading devfile', devfile, 'with optional .che/che-theia-plugins.yaml', cheTheiaPluginsContent, 'and .vscode/extensions.json', vscodeExtensionsJsonContent, 'with sidecar policy', sidecarPolicy);
+    // call library to update devWorkspace and add optional templates
+    await cheTheiaPluginsDevfileResolver.handle({
+      devfile,
+      cheTheiaPluginsContent,
+      vscodeExtensionsJsonContent,
+      devWorkspace,
+      devWorkspaceTemplates,
+      sidecarPolicy,
+      suffix: workspaceId,
+    });
+
+    await Promise.all(devWorkspaceTemplates.map(async template => {
+      // if no namespace, update it
+      if (!template.metadata?.['namespace']) {
+        const templateMetadata: { namespace?: string } = template.metadata || {};
+        if (!templateMetadata.namespace) {
+          templateMetadata.namespace = namespace;
+        }
+        template.metadata = templateMetadata;
+      }
+      const theiaDWT = await this.dwtApi.create(<IDevWorkspaceTemplate>template);
+      const pluginName = theiaDWT.metadata.name;
+      this.addPlugin(createdWorkspace, pluginName, theiaDWT.metadata.namespace);
```

Comment:
for some reason vsxinstaller did not get ownerRef and is left on the cluster 
![Screenshot_20210609_141744](https://user-images.githubusercontent.com/5887312/121345428-83fc6680-c92d-11eb-8676-84c0faecba79.png)

Reply:
I'm preparing the fix and going to push it after testing

Reply:
fixed with https://github.com/eclipse-che/che-dashboard/pull/266/commits/b8e30b469052c210c61ab5228e104df6a8d1b582